### PR TITLE
Add `make help` tip to docs and remove useless type declare

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
Add `make help` tip to docs and remove useless type declare. I think adding a `make help` prompt to the documentation would be very helpful for contributors to start building mimir.

#### Which issue(s) this PR fixes or relates to

None

### CODE REVIEW

File: docs/internal/contributing/README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ To build:
 make
 ```
 
+You can use `make help` to see the available targets.
```
Comment (pracucci) [inline docs/internal/contributing/README.md:37]: I slightly disagree. The problem is that `make help` is currently unsupported for most targets, so it just shows few of them:

```
Usage:
  make <target>

Targets:
  help                                           Display this help and any documented user-facing targets. Other undocumented targets may be present in the Makefile.
  doc                                            Generates the config file documentation.
  dist                                           Generates binaries for a Mimir release.
  format-makefiles                               Format all Makefiles.
  check-doc-validator                            Check documentation using doc-validator tool
  mixin-serve                                    Runs Grafana (listening on port 3000) loading the mixin dashboards compiled at operations/mimir-mixin-compiled.
  mixin-screenshots                              Generates mixin dashboards screenshots.
  docs                                           Serve documentation locally.
```

Giving this suggestion makes the reader think these are just the only available targets, but that's not the case. In particular, this section is "Building Grafana Mimir" and no building target is displayed with `make help`.

I would rather open an issue to document all targets in `make help` first.

Comment (hi-rustin) [reply docs/internal/contributing/README.md:37]: Makes sense.

Comment (hi-rustin) [reply docs/internal/contributing/README.md:37]: I opened an issue: https://github.com/grafana/mimir/issues/2499

Comment (hi-rustin) [reply docs/internal/contributing/README.md:37]: @pracucci I've added help for almost all targets, can we merge it now?

**Review by pracucci — APPROVED** [review]
Sure, we can merge!

**Comment by hi-rustin** [comment]
Thanks for your review and help!  💚 💙 💜 💛 ❤️

**Pull request merged**